config/routes.rb

```diff
@@ -1,10 +1,16 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  get "users/:id" => "users#show", as: :user
+  get "users/:user_id/posts" => "posts#index", as: :user_posts
+  get "users/:user_id/posts/:id" => "posts#show", as: :user_post
+
+
+
+  resources :users, only: [:index, :show] do
+  resources :posts, only: [:index, :show], module: :users
+  end
```

Comment:
- [ ] @NoelLincoln Great work implementing the changes requested, however as the previous reviewer noted you have created your route twice, I would advise you to remove the duplicate code and maybe use the resources for your routes. You can use the suggested code below
```suggestion
  resources :users, only: [:index, :show] do
        resources :posts, only: [:index, :show]
  end
   root "users#index"
```